include/natalie/integer_object.hpp

```diff
@@ -13,6 +14,21 @@
 
 namespace Natalie {
 
+constexpr long long lowest_squarable_fixnum() {
+    nat_int_t mask = (nat_int_t)1 << (sizeof(nat_int_t) * 8 - 2);
+    nat_int_t max_sqrt = 0;
+    while (mask > 0) {
+        nat_int_t current_sum = max_sqrt + mask;
+        mask >>= 1;
+        if ((NAT_MAX_FIXNUM / current_sum) < current_sum) continue; // overflow
+        max_sqrt = current_sum;
+    }
+
+    return max_sqrt;
+}
+
+constexpr long long NAT_MAX_FIXNUM_SQRT = lowest_squarable_fixnum();
```

Comment:
Why not adding this in constants?

Reply:
I did not want to include all the stuff necessary to use nat_int_t in constants.hpp. Also it is not needed for numbers.rb so I thought it might be better to move it. Do you think we should embrace constants.hpp including all the natalie stuff?

(I see I should make NAT_MAX_FIXNUM_SQRT a nat_int_t as well)

Reply:
I am not that strongly for it, I just think it's kind of nice to keep everything in one place, but I am totally cool if we separate them. I also don't really see any use for it outside of nat's runtime, which otherwise would have been a big point for putting it into constants.hpp. I'll leave it to you :D